Input string was not in a correct format. on 5.6.10

Pipeline started to give "Input string was not in a correct format." after the upgrade to 5.6.10:

```
  INFO [05/27/21 14:05:31:15] Working directory: /opt/atlassian/pipelines/agent/build/API/src/API.Library
  INFO [05/27/21 14:05:31:19] Project root is: /opt/atlassian/pipelines/agent/build/
  INFO [05/27/21 14:05:31:19] DotGit directory is: /opt/atlassian/pipelines/agent/build/.git
  INFO [05/27/21 14:05:31:23] Begin: Loading version variables from disk cache
    INFO [05/27/21 14:05:31:24] Cache file /opt/atlassian/pipelines/agent/build/.git/gitversion_cache/36519617206DA1B2C69705907C0975078EFB7D08.yml not found.
    INFO [05/27/21 14:05:31:24] End: Loading version variables from disk cache (Took: 1.68ms)
    INFO [05/27/21 14:05:31:38] Using latest commit on specified branch
    INFO [05/27/21 14:05:33:44] Running against branch: develop (836e82a MGT-1946 Ajustes adicionais nas procs de notificaÁ„o)
    INFO [05/27/21 14:05:33:45] Begin: Calculating base versions
      INFO [05/27/21 14:05:33:63] Fallback base version: 0.1.0 with commit count source 392a317e5ddff35c8bfad2e300aa8446e481fa3f
      INFO [05/27/21 14:05:33:82] Found commit [836e82a MGT-1946 Ajustes adicionais nas procs de notificaÁ„o] matching merge message format: Default
      INFO [05/27/21 14:05:33:82] Found commit [836e82a MGT-1946 Ajustes adicionais nas procs de notificaÁ„o] matching merge message format: Default
      INFO [05/27/21 14:05:33:85] Found commit [836e82a MGT-1946 Ajustes adicionais nas procs de notificaÁ„o] matching merge message format: RemoteTracking
      INFO [05/27/21 14:05:33:86] Found commit [836e82a MGT-1946 Ajustes adicionais nas procs de notificaÁ„o] matching merge message format: RemoteTracking
      INFO [05/27/21 14:05:33:86] Found commit [836e82a MGT-1946 Ajustes adicionais nas procs de notificaÁ„o] matching merge message format: RemoteTracking
      INFO [05/27/21 14:05:33:86] Merge message 'Merge branch 'release/1.82'
  
  # Conflicts:
  #	Migration/src/Migration/Migration.csproj': 1.82.0 with commit count source 1d8f57f17d13af8c247d5cb39183111f17fe4908
      INFO [05/27/21 14:05:33:86] Merge message 'Merge branch 'release/1.74'': 1.74.0 with commit count source b7a311e8b9ef79a6d586387529c769e497105a4f
      INFO [05/27/21 14:05:33:86] Merge message 'Merge remote-tracking branch 'origin/release/1.62' into release/1.62': 1.62.0 with commit count source efaf3d9dea5d56145a33f792c7ea35d51c0c9c0a
      INFO [05/27/21 14:05:33:86] Merge message 'Merge remote-tracking branch 'origin/release/1.54' into release/1.54': 1.54.0 with commit count source c963a066165079e570b125f1dc3aefdb11c9998b
      INFO [05/27/21 14:05:33:86] Merge message 'Merge remote-tracking branch 'origin/release/1.54' into release/1.54
  
  # Conflicts:
  #	FIPEUpdater/UpdaterService.cs': 1.54.0 with commit count source ae293f17ec572943c193084ee835a914ca06b4d8
    INFO [05/27/21 14:05:33:87] End: Calculating base versions (Took: 427.30ms)
    ERROR [05/27/21 14:05:33:89] An unexpected error occurred:
  System.FormatException: Input string was not in a correct format.
     at System.Number.ThrowOverflowOrFormatException(ParsingStatus status, TypeCode type)
     at System.Number.ParseInt32(ReadOnlySpan`1 value, NumberStyles styles, NumberFormatInfo info)
     at System.Int32.Parse(String s)
     at GitVersion.SemanticVersion.TryParse(String version, String tagPrefixRegex, SemanticVersion& semanticVersion) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\SemanticVersioning\SemanticVersion.cs:line 173
     at GitVersion.RepositoryStore.GetValidVersionTags(String tagPrefixRegex, Nullable`1 olderThan) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\RepositoryStore.cs:line 400
     at GitVersion.VersionCalculation.TaggedCommitVersionStrategy.GetTaggedVersions(IBranch currentBranch, Nullable`1 olderThan) in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculators\TaggedCommitVersionStrategy.cs:line 29
     at GitVersion.VersionCalculation.TaggedCommitVersionStrategy.GetVersions() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculators\TaggedCommitVersionStrategy.cs:line 24
     at GitVersion.VersionCalculation.BaseVersionCalculator.GetBaseVersions(IVersionStrategy strategy)+MoveNext() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculator.cs:line 90
     at System.Collections.Generic.List`1..ctor(IEnumerable`1 collection)
     at System.Linq.Enumerable.ToList[TSource](IEnumerable`1 source)
     at GitVersion.VersionCalculation.BaseVersionCalculator.GetBaseVersion() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\BaseVersionCalculator.cs:line 31
     at GitVersion.VersionCalculation.NextVersionCalculator.FindVersion() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\VersionCalculation\NextVersionCalculator.cs:line 58
     at GitVersion.GitVersionCalculateTool.CalculateVersionVariables() in D:\a\GitVersion\GitVersion\src\GitVersion.Core\Core\GitVersionCalculateTool.cs:line 52
     at GitVersion.GitVersionExecutor.RunGitVersionTool(GitVersionOptions gitVersionOptions) in D:\a\GitVersion\GitVersion\src\GitVersion.App\GitVersionExecutor.cs:line 70
    INFO [05/27/21 14:05:33:89] Attempting to show the current git graph (please include in issue): 
    INFO [05/27/21 14:05:33:89] Showing max of 100 commits
    INFO [05/27/21 14:05:33:95] * 92c1cdf71 11 minutes ago  (origin/feature/MGT-1944-faturamento-de-grupo-economico)
  | * 836e82a46 29 minutes ago  (HEAD -> develop, origin/develop)
  | *   667dbfe19 13 hours ago 
  | |\  
  | | * bd7c11559 2 days ago  (tag: v1.108.1, origin/master, origin/HEAD, master)
  | | * a99edb242 3 days ago  (tag: v1.108.0, origin/release/1.108, release/1.108)
  | | * 29014ce9c 3 days ago 
  | | * 742699dae 3 days ago 
  | | * 5453d11b2 3 days ago 
  | | * b1e424ace 3 days ago 
  | | * d185b26ff 3 days ago 
  | | * 692917738 3 days ago 
  | * |   8c66fc27a 13 hours ago 
  | |\ \  
  | | * | 84d567320 13 hours ago  (origin/feature/MGT-1946-melhorias-nas-notificaÁıes)
  | * | |   05381d82a 13 hours ago 
  | |\ \ \  
  | | |/ /  
  | |/| |   
  | | * | 279c8e147 3 days ago 
  | | * | 815112144 3 days ago 
  | | |/  
  | * | aba256eff 2 days ago 
  | * | 8b5794c4c 2 days ago 
  |/ /  
  | | * 032826d76 2 days ago  (origin/feature/MGT-1837-permitir-ediÁ„o-de-campos-da-oc)
  | |/  
  |/|   
  * | 061563226 3 days ago  (origin/feature/MGT-1910-integraÁ„o-vexsoft-checklist)
  * | efa203a5d 3 days ago 
  |/  
  * 59e5c03e2 4 days ago  (tag: v1.108.0-rc1)
  *   f852d05ec 4 days ago 
  |\  
  | * 2d9eb05f1 4 days ago 
  | * 188e67f97 5 days ago 
  | * 96a70cfcb 5 days ago 
  | * cd7d84c57 6 days ago 
  | * 3a5eb9567 6 days ago 
  | * a1539efd0 7 days ago 
  | * 52f07e58d 7 days ago 
  * |   223927da2 5 days ago 
  |\ \  
  | * | 73845a379 5 days ago 
  | * |   306994f91 5 days ago 
  | |\ \  
  | | * | 20a66177d 5 days ago 
  | | * |   a547c0f27 5 days ago 
  | | |\ \  
  | |_|/ /  
  |/| | |   
  * | | | bd922c7c1 5 days ago 
  * | | | 587f0d533 6 days ago 
  * | | |   0febe7a63 7 days ago 
  |\ \ \ \  
  | | |_|/  
  | |/| |   
  | * | |   cbd133564 7 days ago 
  | |\ \ \  
  | * | | | 2508e2e8a 7 days ago 
  | * | | | 9b8b4f099 7 days ago 
  | * | | | 70e6f5912 8 days ago 
  | * | | | 7c7cab55e 8 days ago 
  | * | | | faab5bfab 9 days ago 
  | * | | | 1bd80a038 9 days ago 
  | * | | | 6128b56d4 9 days ago 
  | * | | | 67de79c3b 9 days ago 
  | * | | | 7a36116e6 10 days ago 
  | * | | | d97155da4 10 days ago 
  * | | | |   2a436fdd8 7 days ago 
  |\ \ \ \ \  
  | |_|/ / /  
  |/| | | |   
  | * | | | a8844e750 7 days ago 
  | * | | | 9d100a80a 9 days ago 
  | * | | | 4fb65f478 9 days ago 
  | * | | | 85a1a2c9c 11 days ago 
  | * | | | 9168627d8 11 days ago 
  | * | | | 2856f6d8e 13 days ago 
  | * | | | c8acf0ac6 2 weeks ago 
  | * | | | c1e14fa9b 3 weeks ago 
  * | | | |   293262f72 8 days ago 
  |\ \ \ \ \  
  | * | | | | a1c439b5f 8 days ago  (origin/feature/MGT-1921-procs-implementar-notificaÁıes)
  | * | | | | aedde0732 8 days ago 
  | * | | | | d45a44408 8 days ago 
  | * | | | | 837d0de22 9 days ago 
  | * | | | | 32203018a 10 days ago 
  * | | | | |   e8f02e803 8 days ago 
  |\ \ \ \ \ \  
  | * | | | | | 84c634688 8 days ago 
  | * | | | | | 22c7d91aa 9 days ago 
  | * | | | | | e3f63abfc 10 days ago 
  | | |_|/ / /  
  | |/| | | |   
  | * | | | | dc4d446f7 10 days ago 
  | * | | | |   3057590ac 10 days ago 
  | |\ \ \ \ \  
  | | |/ / / /  
  | |/| | | |   
  | | * | | | 6ee862b02 10 days ago  (origin/feature/MGT-1905-permitir-aplicaÁ„o-do-reajuste-)
  * | | | | | 2e572c36c 9 days ago 
  * | | | | |   56484da7e 9 days ago 
  |\ \ \ \ \ \  
  | |/ / / / /  
  |/| | | | |   
  | * | | | | 94bf77926 9 days ago 
  | * | | | | e6f289fce 9 days ago 
  | * | | | | 488bf6b90 10 days ago 
  |/ / / / /  
  | | | * | 95289b651 5 days ago 
  | | | * | 7e02263a4 5 days ago 
  | | | |/  
  | | | * ea4bdef4e 6 days ago 
  | | | * 2992a57c5 6 days ago 
  | | | * 41c5e560e 6 days ago 
  | | | * 4132546a8 6 days ago 
  | | | * 1cfdada7a 7 days ago 
  | | | * 815102054 7 days ago 
  | | | * c6dd41710 8 days ago 
  | | | * 5b9de0846 8 days ago 
  | | | * a491c1ecb 8 days ago 
  | | | * b01f202ab 9 days ago 
  | |_|/  
  |/| |   
  * | |   cc3df895e 11 days ago 
  |\ \ \  
  | * | | f9160d2b3 11 days ago 
  |/ / /  
  * | |   133bcde0a 12 days ago 
  |\ \ \  
  | * | | b793ab52d 12 days ago  (origin/feature/MGT-1909-bug-quando-altera-o-entity-zera)
  | * | | 898b78b94 12 days ago 
  | * | | b2cf4ec6f 12 days ago 
  |/ / /  
  * | |   87b688ca9 12 days ago 
  |\ \ \  
  | * | | 9619b5628 13 days ago 
  | * | |   e07d0dd03 13 days ago 
  | |\ \ \  
  | |/ / /  
  |/| / /   
  | |/ /    
  * | |   a62d4ba6d 13 days ago 
  |\ \ \  
  | * | | 4583ffab4 2 weeks ago 
  * | | |   40eb6e326 13 days ago 
  |\ \ \ \  
  
/opt/atlassian/pipelines/agent/build/API/src/API.Library/API.Library.csproj(13,5): error MSB3073: The command "dotnet-gitversion /showvariable FullSemVer" exited with code 1.

Build FAILED.

/opt/atlassian/pipelines/agent/build/API/src/API.Library/API.Library.csproj(13,5): error MSB3073: The command "dotnet-gitversion /showvariable FullSemVer" exited with code 1.
    0 Warning(s)
    1 Error(s)
```

--- a/src/GitVersion.Core.Tests/VersionCalculation/SemanticVersionTests.cs
+++ b/src/GitVersion.Core.Tests/VersionCalculation/SemanticVersionTests.cs
@@ -57,9 +57,10 @@ namespace GitVersion.Core.Tests
 
         [TestCase("someText")]
         [TestCase("some-T-ext")]
-        public void ValidateInvalidVersionParsing(string versionString)
+        [TestCase("v.1.2.3", "v")]
+        public void ValidateInvalidVersionParsing(string versionString, string tagPrefixRegex = null)
         {
-            Assert.IsFalse(SemanticVersion.TryParse(versionString, null, out _), "TryParse Result");
+            Assert.IsFalse(SemanticVersion.TryParse(versionString, tagPrefixRegex, out _), "TryParse Result");
         }
 
         [Test]

--- a/src/GitVersion.Core/VersionCalculation/SemanticVersioning/SemanticVersion.cs
+++ b/src/GitVersion.Core/VersionCalculation/SemanticVersioning/SemanticVersion.cs
@@ -8,7 +8,7 @@ namespace GitVersion
         private static SemanticVersion Empty = new SemanticVersion();
 
         private static readonly Regex ParseSemVer = new Regex(
-            @"^(?<SemVer>(?<Major>\d+)?(\.(?<Minor>\d+))?(\.(?<Patch>\d+))?)(\.(?<FourthPart>\d+))?(-(?<Tag>[^\+]*))?(\+(?<BuildMetaData>.*))?$",
+            @"^(?<SemVer>(?<Major>\d+)(\.(?<Minor>\d+))?(\.(?<Patch>\d+))?)(\.(?<FourthPart>\d+))?(-(?<Tag>[^\+]*))?(\+(?<BuildMetaData>.*))?$",
             RegexOptions.Compiled);
 
         public int Major;

